src/layman/user/rest_user.py

```diff
@@ -0,0 +1,58 @@
+from flask import current_app as app, g, Blueprint, jsonify, request
+from layman.authn import authenticate
+from layman import util as layman_util
+from layman.layer import LAYER_TYPE, util as layer_util
+from layman.map import MAP_TYPE, util as map_util
+from .util import is_allowed_to_delete_user, get_user_profile, delete_user
+
+
+bp = Blueprint('rest_user', __name__)
+
+
+@bp.before_request
+@authenticate
+def before_request():
+    pass
+
+
+@bp.route('/<username>', methods=['DELETE'])
+def delete(username):
+    app.logger.info(f"DELETE User request for username={username} by user={g.user}")
+    if username not in layman_util.get_usernames(use_cache=False):
+        app.logger.warning(f"User {username} does not exist.")
+        return jsonify({'error': 'User not found'}), 404
+    user_profile = get_user_profile(g.user)
+    user_profile_name = user_profile.get('username')
+    if not user_profile_name:
+        return jsonify({'error': 'User profile is missing username'}), 400
+    is_authorized = is_allowed_to_delete_user(user_profile_name)
```

Comment:
Please focus only on self-deletion in this PR. Keep deleting other users for another PR. So remove `is_allowed_to_delete_user` call.